src/appleseed/renderer/kernel/shading/shadingpoint.h

```diff
@@ -114,8 +122,15 @@ class ShadingPoint
     // Return the time stored in the ray.
     const ShadingRay::Time& get_time() const;
 
-    // Return true if an intersection was found, false otherwise.
-    bool hit() const;
+    // Return true if the shading point is either
+    // a surface point or a volume point and can be shaded.
+    bool valid() const;
```

Comment:
I would recommend `is_valid()`. The only other place where we have a `valid()` method is on `foundation::CDF` but I'll change it to `is_valid()` as well.